clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -576,16 +590,21 @@ private void addToCompletedReceives() {
                 KafkaChannel channel = entry.getKey();
                 if (!channel.isMute()) {
                     Deque<NetworkReceive> deque = entry.getValue();
-                    NetworkReceive networkReceive = deque.poll();
-                    this.completedReceives.add(networkReceive);
-                    this.sensors.recordBytesReceived(channel.id(), networkReceive.payload().limit());
+                    addToCompletedReceives(channel, deque, false);
```

Comment:
Sorry, I'm too used to thinking from the client perspective. I was wondering why it made sense for clients. After talking with @ijuma, it seems like the intent is perhaps to keep the cost of poll() down.